tests/jobs/test_backfill_job.py

```diff
@@ -147,31 +147,22 @@ def test_dag_run_with_finished_tasks_set_to_success(self, dag_maker):
 
         assert State.SUCCESS == dag_run.state
 
-    @pytest.mark.xfail(condition=True, reason="This test is flaky")
     @pytest.mark.backend("postgres", "mysql")
     def test_trigger_controller_dag(self):
         dag = self.dagbag.get_dag("example_trigger_controller_dag")
         target_dag = self.dagbag.get_dag("example_trigger_target_dag")
         target_dag.sync_to_db()
 
-        # dag_file_processor = DagFileProcessor(dag_ids=[], log=Mock())
-        task_instances_list = []
-        # task_instances_list = dag_file_processor._process_task_instances(
-        #    target_dag,
-        #    dag_runs=DagRun.find(dag_id='example_trigger_target_dag')
-        # )
-        assert not task_instances_list
+        target_dag_run = target_dag.get_last_dagrun()
```

Comment:
Instead of relying on `get_last_dagrun()` (which can have bugs), it could be better to use SQLAlchemy directly here. There is a `session` pytest fixture you can use for this.